src/app/controllers/recording_controller.py

```diff
@@ -0,0 +1,126 @@
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot
+
+from src.app.services.recorder.recorder import Recorder, RecorderActions
+from src.app.services.videoprocessing.video_processor import VideoProcessor
+
+
+class RecordingController(QObject):
+
+    signalException = pyqtSignal(Exception)
+    signalOdasState = pyqtSignal(bool)
+    signalRecordingState = pyqtSignal(bool)
+    signalVideoProcessorState = pyqtSignal(bool)
+    signalVirtualCamerasReceived = pyqtSignal(object, object)
+
+    def __init__(self, outputFolder, odasserver, parent=None):
+        super(RecordingController, self).__init__(parent)
+        self.__odasserver = odasserver
+
+        self.__recorder = Recorder(outputFolder)
+        self.__recorder.changeAudioSettings(outputFolder=outputFolder, nChannels=4, nChannelFile=1, byteDepth=2, sampleRate=48000)
+        self.__recorder.start()
+
+        self.isRecording = False
+        self.isOdasLiveConnected = False
+        self.videoProcessorState = False
+
+        self.__recorder.signalException.connect(self.exceptionHandling)
+
+        self.__odasserver.signalException.connect(self.exceptionHandling)
+        self.__odasserver.signalAudioData.connect(self.audioDataReceived)
+        self.__odasserver.signalClientsConnected.connect(self.odasClientConnected)
+
+        self.__videoProcessor = VideoProcessor()
+        self.__videoProcessor.signalException.connect(self.exceptionHandling)
+        self.__videoProcessor.signalFrameData.connect(self.virtualCamerasReceived)
```

Comment:
signalVirtualCameras, and should be connected only when the videoprocessor is running in the current controller